phys/module_surface_driver.F

```diff
@@ -3863,6 +3869,212 @@ SUBROUTINE surface_driver(                                         &
 ! -------------------------------------------------------------------
 #endif
 
+#ifdef WRF_USE_CTSM
+     CASE (CTSMSCHEME)
+
+! TODO (NS): Adding errors and warning on compatible PBL schemes, etc. For
+! example:                                                                                                  
+     IF (MYJ) call wrf_error_fatal('CLM is not currently compatible with MYJ.  Please pick a different PBL scheme,')
+
+     IF (PRESENT(qv_curr)    .AND.  PRESENT(rainbl)        .AND.  &
+       ! PRESENT(emiss)      .AND.  PRESENT(t2)            .AND.    &
+       ! PRESENT(declin) .AND.  PRESENT(coszen)    .AND.            &
+       ! PRESENT(hrang)  .AND. PRESENT( xlat_urb2d)    .AND.        &
+       ! PRESENT(dzr)       .AND.                                   &
+       ! PRESENT( dzb)            .AND. PRESENT(dzg)       .AND.    &
+       ! PRESENT(tr_urb2d) .AND. PRESENT(tb_urb2d)         .AND.    &
+       ! PRESENT(tg_urb2d) .AND. PRESENT(tc_urb2d) .AND.            &
+       ! PRESENT(qc_urb2d) .AND. PRESENT(uc_urb2d) .AND.            &
+       ! PRESENT(xxxr_urb2d) .AND. PRESENT(xxxb_urb2d) .AND.        &
+       ! PRESENT(xxxg_urb2d) .AND.                                  &
+       ! PRESENT(xxxc_urb2d) .AND. PRESENT(trl_urb3d) .AND.         &
+       ! PRESENT(tbl_urb3d)   .AND. PRESENT(tgl_urb3d)  .AND.       &
+       ! PRESENT(sh_urb2d) .AND. PRESENT(lh_urb2d)  .AND.           &
+       ! PRESENT(g_urb2d)   .AND. PRESENT(rn_urb2d) .AND.           &
+       ! PRESENT(ts_urb2d)                          .AND.           &
+       ! PRESENT(frc_urb2d) .AND. PRESENT(utype_urb2d)   .AND.      &
+#if (EM_CORE==1)         
+         PRESENT (smcwtdxy)       .AND.                              &
+         PRESENT (rechxy)         .AND.                              &
+         PRESENT (deeprechxy)     .AND.                              &
+         PRESENT (fdepthxy)       .AND.                              &
+         PRESENT (areaxy)         .AND.                              &
+         PRESENT (rivercondxy)    .AND.                              &
+         PRESENT (riverbedxy)     .AND.                              &
+         PRESENT (eqzwt)          .AND.                              &
+         PRESENT (pexpxy)         .AND.                              &
+         PRESENT (qrfxy)          .AND.                              &
+         PRESENT (qspringxy)      .AND.                              &
+         PRESENT (qslatxy)        .AND.                              &
+         PRESENT (qrfsxy)         .AND.                              &
+         PRESENT (qspringsxy)     .AND.                              &
+         PRESENT (smoiseq)        .AND.                              &
+         PRESENT (wtddt)          .AND.                              &
+         PRESENT (stepwtd)        .AND.                              &
+#endif                   
```

Comment:
What's the reason to check the presence of these variables?

Reply:
@weiwangncar , you are absolutely right. Checking if these variables are present is not necessary at all. I copied this from NOAH-MP, as I thought we might want to use similar to NOAH-MP handling of groundwater in here. But we don't need this here at all. I removed them.